### PR TITLE
enable-flatcar-os-bastion

### PR DESCRIPTION
**Pull request by calvix**
use CAPI bootstrap controller to manage flatcar for bastion node
towards https://github.com/giantswarm/roadmap/issues/996

### CODE REVIEW

File: helm/cluster-aws/files/etc/ssh/sshd_config [hunk 1]
```diff
@@ -13,3 +13,4 @@ MaxAuthTries 5
 LoginGraceTime 60
 AllowTcpForwarding no
 AllowAgentForwarding no
+CASignatureAlgorithms ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-ed25519,rsa-sha2-512,rsa-sha2-256,ssh-rsa
```
Comment (calvix) [inline helm/cluster-aws/files/etc/ssh/sshd_config:16]: still necessary as the flatcar is denying RSA by default

File: helm/cluster-aws/templates/_bastion.tpl [hunk 2]
```diff
@@ -99,4 +87,31 @@ metadata:
   name: {{ include "resource.default.name" $ }}-bastion-{{ include "hash" (dict "data" (include "bastion-awsmachinetemplate-spec" $) "global" .) }}
   namespace: {{ .Release.Namespace }}
 spec: {{ include "bastion-awsmachinetemplate-spec" $ | nindent 2 }}
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+kind: KubeadmConfigTemplate
```
Comment (fiunchinho) [inline helm/cluster-aws/templates/_bastion.tpl:92]: Has this object a similar immutability to other CAPA `*Template` CRs like `AWSMachineTemplate`? For those we create a different CR when changing it. Is that necessary here?

Comment (calvix) [reply helm/cluster-aws/templates/_bastion.tpl:92]: Well we do not have it for  `KubeadmConfig` for the MachinePool but I found something similar for the MachineDeployment in ` cluster-gcp` so maybe its worth integrating similar approach

Comment (fiunchinho) [reply helm/cluster-aws/templates/_bastion.tpl:110]: I would only do it if there are immutable fields, otherwise it's just adding more complexity.

Comment (calvix) [reply helm/cluster-aws/templates/_bastion.tpl:110]: I think it is the case as we needed it for the MD in gcp

**Review by fiunchinho — APPROVED** [review]

**Pull request merged**